lang/java/reef-io/src/main/java/org/apache/reef/io/network/group/impl/config/GatherOperatorSpec.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.io.network.group.impl.config;
+
+import org.apache.reef.io.network.group.api.config.OperatorSpec;
+import org.apache.reef.io.network.group.impl.utils.Utils;
+import org.apache.reef.io.serialization.Codec;
+
+/**
+ * The specification for the Gather operator.
+ */
+public class GatherOperatorSpec implements OperatorSpec {
+
+  private final String receiverId;
+
+  /**
+   * Codec to be used to serialize data.
+   */
+  private final Class<? extends Codec> dataCodecClass;
+
+  public GatherOperatorSpec(final String receiverId,
+                            final Class<? extends Codec> dataCodecClass) {
+    super();
```

Comment:
I added `super()` to the constructor of this class and `ScatterOperatorSpec` because the other `OperatorSpec` classes did too, but since these classes are not extending another concrete class or abstract class, I guess we can just take it out. I'll remove it.